src/test/lib/ddl/ddl_statement_test.cpp

```diff
@@ -0,0 +1,236 @@
+    auto actual_indices = current_chunk->get_indexes(*column_ids);
+    EXPECT_TRUE(actual_indices.size() == static_cast<unsigned long>(index_count));
+  }
+}
+
+TEST_F(DDLStatementTest, CreateIndexSingleColumn) {
+  create_index(_create_index_single_column);
+
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = _table_a->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "myindex");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  check_if_index_exists_correctly(column_ids, _table_a);
+}
+
+TEST_F(DDLStatementTest, CreateIndexMultiColumn) {
+  create_index(_create_index_multi_column);
+
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+  column_ids->emplace_back(ColumnID{1});
+
+  auto actual_index = _table_a->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "myindex");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  check_if_index_exists_correctly(column_ids, _table_a);
+}
+
+TEST_F(DDLStatementTest, CreateIndexWithoutName) {
+  create_index("CREATE INDEX ON table_a (a)");
+
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = _table_a->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "table_a_a");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  check_if_index_exists_correctly(column_ids, _table_a);
+}
+
+TEST_F(DDLStatementTest, CreateIndexIfNotExistsFirstTime) {
+  create_index("CREATE INDEX IF NOT EXISTS myindex ON table_a (a)");
+
+  auto column_ids = std::make_shared<std::vector<ColumnID>>();
+  column_ids->emplace_back(ColumnID{0});
+
+  auto actual_index = _table_a->indexes_statistics().at(0);
+
+  EXPECT_TRUE(actual_index.name == "myindex");
+  EXPECT_TRUE(actual_index.column_ids == *column_ids);
+
+  check_if_index_exists_correctly(column_ids, _table_a);
+}
+
+TEST_F(DDLStatementTest, CreateIndexExistsFlagSet) {
+  create_index(_create_index_single_column);
+
+  auto second_sql_pipeline = SQLPipelineBuilder{"CREATE INDEX IF NOT EXISTS myindex ON table_a (a, b)"}.create_pipeline();
+
+  const auto& [second_pipeline_status, second_table] = second_sql_pipeline.get_result_table();
+  EXPECT_EQ(second_pipeline_status, SQLPipelineStatus::Success);
+
+  auto single_column_col_ids = std::make_shared<std::vector<ColumnID>>();
+  single_column_col_ids->emplace_back(ColumnID{0});
+
+  check_if_index_exists_correctly(single_column_col_ids, _table_a);
+}
+
+
+TEST_F(DDLStatementTest, CreateIndexExistsFlagNotSet) {
+  create_index(_create_index_single_column);
+
+  auto second_sql_pipeline = SQLPipelineBuilder{"CREATE INDEX myindex ON table_a (a, b)"}.create_pipeline();
+
+  // TODO: come up with way to test this without aborting test execution
+  // EXPECT_THROW(second_sql_pipeline.get_result_table(), std::exception);
```

Comment:
We do not see why this should not work. Maybe it fails in a later stage of this test?